src/HotChocolate/Data/src/Data/Projections/Optimizers/IsProjectedProjectionOptimizer.cs

```diff
@@ -17,55 +24,191 @@ public class IsProjectedProjectionOptimizer : IProjectionOptimizer
         SelectionSetOptimizerContext context,
         Selection selection)
     {
-        if (!(context.Type is ObjectType type &&
-                type.ContextData.TryGetValue(AlwaysProjectedFieldsKey, out var fieldsObj) &&
-                fieldsObj is string[] fields))
+        var type = (ObjectType) context.Type;
+        var alwaysProjectedFieldNames = (string[]) type.ContextData[AlwaysProjectedFieldsKey]!;
+
+        int aliasedFieldIndex = 0;
+        string GetCurrentAlias() => _fieldAliasesCache[aliasedFieldIndex];
```

Comment:
Something like this maybe make sense? Otherwise we run the issue to run into a array out of bounds exception
```suggestion
        string GetCurrentAlias() =>  aliasedFieldIndex < _aliasCount 
                ? _fieldAliasesCache[aliasedFieldIndex]
                 : "__projection_alias_" + aliasedFieldIndex`
```

Reply:
Well then let's just make the array dynamic, initialize the string lazily, and double it when it's accessed past the max length. Thread safety shouldn't be an issue, I think.

Reply:
i just dont like having a static array with no boundaries if there is no use for it.
more than 8 really doesnt make much sense for a user. But as we are working on the incoming query here you could exploit it. I am not saying it's possible to to this right now, but having it just capped at 8 make is impossible to exploit , even with a possible future change.

Reply:
Right, thanks